superset-frontend/src/dashboard/components/menu/HoverMenu.tsx

```diff
@@ -81,6 +82,8 @@ export default class HoverMenu extends React.PureComponent<HoverMenuProps> {
             position === 'left' && 'hover-menu--left',
             position === 'top' && 'hover-menu--top',
           )}
+          onMouseEnter={onHover}
+          onMouseLeave={onHover}
```

Comment:
Technically this component knows the state of the hover. OnMouseEnter = hovered onMouseLeave = not hovered. Instead of having the parent storing and toggling the state (which could get out of sync), I think we can be more explicit by letting this HoverMenu component save the state of the hover, and then there's already a ref getting passed in. We should be able to use that ref to get the state of the hover instead of having the onHover callback.

Reply:
@eschutho I've tried to add a boolean property of `hoveredState` to `inneRef.current` dependent on the `onMouseEnter/Leave` handlers soley inside `HoverMenu` but any changes are not registering when trying to add a class to the `GridRow` in `Row.jsx`. It might be that the `innerRef` is to `Draggable` so I would need to create some logic in there to handle the `hoveredState`? I think I need a more detailed rundown of what you mean in this case...

Reply:
@eschutho Yikes, after an extensive rabbit hole it looks like there isn't really a great way of using a ref to update the state of `Row` based on hovered state of `HoverMenu` without a context provider. I'm thinking a callback is the better choice here but just in case, I made a commit with my attempt at handling things with a ref.